crates/adapters/src/controller/mod.rs

```diff
@@ -1037,41 +1033,66 @@ impl ControllerInner {
                 ControllerError::unknown_input_stream(endpoint_name, &endpoint_config.stream)
             })?;
 
-        // Create parser.
-        let format_config = endpoint_config
-            .connector_config
-            .format
-            .as_ref()
-            .ok_or_else(|| ControllerError::input_format_not_specified(endpoint_name))?
-            .clone();
-        let format = <dyn InputFormat>::get_format(&format_config.name).ok_or_else(|| {
-            ControllerError::unknown_input_format(endpoint_name, &format_config.name)
-        })?;
-
-        let parser = format.new_parser(endpoint_name, input_handle, &format_config.config)?;
-
         // Create probe.
         let endpoint_id = inputs.keys().next_back().map(|k| k + 1).unwrap_or(0);
-        let probe = Box::new(InputProbe::new(
-            endpoint_id,
-            endpoint_name,
-            parser,
-            self.clone(),
-            self.circuit_thread_unparker.clone(),
-            self.backpressure_thread_unparker.clone(),
-        ));
 
-        // Initialize endpoint stats.
-        self.status.add_input(
-            &endpoint_id,
-            endpoint_name,
-            endpoint_config,
-            endpoint.is_fault_tolerant(),
-        );
+        let reader = match endpoint {
```

Comment:
> Regarding a connector vs. an integrated connector, is the following understanding correct?
> 
>     * Usually a connector is decoupled from the format. The connector inherently has some "unit" in which its data is separated, it being one unit (e.g., a single file as a whole), or many units (e.g., Kafka messages). This "unit" of data is considered a binary blob that the format turns into tuples the DBSP circuit can ingest.
> 
>     * An integrated connector has a fixed format.

Yep, this is correct.
 
> Would it not be simpler to have a format variant called `Integrated` or `None`, and upon matching to a transport which is integrated and for which the format variant is not that, throw an error?

We could do that, but I don't think this will simplify this code. We need two branches to build two different input pipelines, the "normal' one consisting of `transport -> input_probe -> format parser`, and an integrated one that only has one component (the integrated adapter). Maybe I didn't understand the suggestion. Can you clarify what you had in mind?

Reply:
I am not as intimately familiar with this particular part of the code base; conceptually I imagined the building to be a function which takes as parameters (1) transport configuration and (2) format configuration, and returns a built connector. What happens internally in terms of validation, is up to the implementation of the connector transport (though, probably will make use of similar helper functions). If a combination of (transport, format) which is not valid is given, it returns an error.

I guess it does not matter as the API does not differ with the implementation, as such whichever works best in current code base sounds good. What happens currently if a format is provided even though the transport does not support it?

Reply:
Ok, I see what you mean now. The advantage of the current way of doing this is that non-integrated connectors don't even need to know that data formats exist. But I think we could still implement your suggestion by having a special integrated connector implementation act as a wrapper around "regular" connectors and do the check. That would be cleaner, but I probably won't do this in this PR though.

> What happens currently if a format is provided even though the transport does not support it?

Good point, it will be ignored, I'll fix this.